docs/features/subclients.md

```diff
@@ -0,0 +1,197 @@
+## Operation groups
+
+autorest requires each operation in swagger file to define a unique string parameter `operationId` (this is different from [official OpenAPI specification](https://swagger.io/docs/specification/paths-and-operations/#operationId)). If parameter value contains underscore, then part after underscore will be treated as operation name, and part before underscore will be treated as the name of the operation group to which this operation belongs. Otherwise, operation will be attributed to the group without a name::
+
+```js
+"paths": {
+  "/namedgroup/op": {
+    "get": {
+      "operationId": "NamedGroup_Op1",
+    }
+  },
+  "/nogroup/op": {
+    "get": {
+      "operationId": "Op2",
+    }
+  }
+}
+```
+
+```yaml
+operationGroups:
+  - !OperationGroup 
+    $key: NamedGroup
+    operations:
+      - !Operation 
+        language: !Languages 
+          default:
+            name: Op1
+    language: !Languages 
+      default:
+        name: NamedGroup
+  - !OperationGroup 
+    $key: ''
+    operations:
+      - !Operation 
+        language: !Languages 
+          default:
+            name: Op2
+    language: !Languages 
+      default:
+        name: ''
+language: !Languages 
+  default:
+    name: MyService
+```
+
+For each operation group, **autorest.csharp** generates individual client type called using `$"{operationGroups.language.default.name}Client"` format when group has a name or `$"{language.default.name}Client"` format when group name is empty.
+
+```cs
+public partial class NamedGroupClient
+{
+    public virtual async Task<Response> Op1Async(RequestOptions options = null);
```

Comment:
`RequestOptions` got renamed to `RequestContext`. Can we rename it everywhere?

Reply:
Fixed